File: app/src/main/java/yoyo/app/android/com/yoyoapp/Trip/booking/BookingSecondFragment.java
package yoyo.app.android.com.yoyoapp.Trip.booking;

import android.content.Context;
import android.net.Uri;
import android.os.Bundle;
import android.widget.TextView;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import yoyo.app.android.com.yoyoapp.FragmentTransaction.BaseFragment;
import yoyo.app.android.com.yoyoapp.R;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class BookingSecondFragment extends BaseFragment {


    private View view;
    private TextView orderDateTextView, customerNameTextView, quantityTextView, totalPriceTextview;
    private int travellerNum;
    private double price;
    private String url,name,date;


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.fragment_booking_second2, container, false);

        init();
        getBundle();
        calculatePrice();
        setupViews();

        return view;
    }

    private void setupViews() {
        totalPriceTextview.setText(String.valueOf(price));
        customerNameTextView.setText(name);
        quantityTextView.setText(String.valueOf(travellerNum));
        orderDateTextView.setText(date);
    }

    private void calculatePrice() {
        price = price * travellerNum;
        date = new SimpleDateFormat("yyyy-MM-dd", Locale.getDefault()).format(new Date());
    }

    private void getBundle() {
        Bundle bundle = getArguments();
        travellerNum = bundle.getInt("travellerNumber");
        price = bundle.getDouble("price");
        url = bundle.getString("url");
        name = bundle.getString("name");
    }

    private void init() {
        orderDateTextView = view.findViewById(R.id.tv_booking_second_order_date);
        customerNameTextView = view.findViewById(R.id.tv_booking_second_customer_name);
        quantityTextView = view.findViewById(R.id.tv_booking_second_quantity);
        totalPriceTextview = view.findViewById(R.id.tv_booking_second_total_price);
    }


}

File: app/src/main/java/yoyo/app/android/com/yoyoapp/trip/booking/BookingSecondFragment.kt
package yoyo.app.android.com.yoyoapp.trip.booking

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_booking_second2.view.*

import yoyo.app.android.com.yoyoapp.FragmentTransaction.BaseFragment
import yoyo.app.android.com.yoyoapp.R
import yoyo.app.android.com.yoyoapp.trip.dialog.ExitDialogFragment

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class BookingSecondFragment : BaseFragment() {

    private var travellerNum: Int = 0
    private var price: Double = 0.toDouble()
    private var url: String? = null
    private var name: String? = null
    private var date: String? = null


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val res = inflater.inflate(R.layout.fragment_booking_second2, container, false)

        getBundle()
        calculatePrice()
        setupViews(res)
        res.button_booking_pay.setOnClickListener{ setupPay()}
        res.iv_sign_out_back.setOnClickListener{setupExit()}

        return res
    }

    private fun setupExit() {
        val exitDialogFragment = ExitDialogFragment()
        fragmentManager?.let { exitDialogFragment.show(it, "exit") }
    }

    private fun setupPay() {
        view?.progressbar_booking_book?.visibility = View.VISIBLE
        val i = Intent(Intent.ACTION_VIEW)
        i.data = Uri.parse(url)
        startActivity(i)
        view?.progressbar_booking_book?.visibility = View.GONE
    }

    private fun setupViews(res: View) {
        res.tv_booking_second_total_price.text = price.toString()
        res.tv_booking_second_customer_name.text = name
        res.tv_booking_second_quantity.text = travellerNum.toString()
        res.tv_booking_second_order_date.text = date
    }

    private fun calculatePrice() {
        price *= travellerNum
        date = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault()).format(Date())
    }

    private fun getBundle() {
        arguments?.apply {
            travellerNum = getInt("travellerNumber")
            price = getDouble("price")
            url =  getString("url")
            name = getString("name")
        }

    }

}
